rs_zktrie/src/raw.rs

```diff
@@ -634,7 +634,7 @@ impl<H: Hashable, DB: ZktrieDatabase, const MAX_LEVELS: usize> ZkTrieImpl<H, DB,
                     )?;
                     // go up until the root
                     final_root = Some(self.recalculate_path_until_root(
-                        &path_remain,
+                        path_remain.clone(),
```

Comment:
Why we even use a clone here

Reply:
It's an iterator, clone is cheap and it will be consumed